cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/main/SampledLogging.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.report.main;
+
+import org.slf4j.Logger;
+
+import javax.annotation.Nullable;
+
+/**
+ * log warning with interval count
+ */
+public class SampledLogging {
```

Comment:
You can get a sampling log with Loggers.sampling without implementing this

Reply:
that's in cdap-common, we cant depend on that.